### PR TITLE
Clang prefix on Darwin

### PR DESCRIPTION
**Pull request by DarthGandalf**
Bug: https://bugs.gentoo.org/758167

Perhaps the patch should be included into mgorny's patchset instead

I'm making this flag global because quite a few LLVM packages require this, and there will be one more (sys-devel/clang-common), which is not in this PR.

### CODE REVIEW

File: eclass/llvm.eclass [hunk 1]
```diff
@@ -64,6 +64,8 @@ esac
 if [[ ! ${_LLVM_ECLASS} ]]; then
 _LLVM_ECLASS=1
 
+IUSE="bootstrap-prefix"
```
Comment (mgorny) [inline eclass/llvm.eclass:67]: It sounds like a serious overkill to add a bootstrap flag to all packages using LLVM in any way.

**Review by mgorny — CHANGES_REQUESTED** [review]

File: eclass/llvm.eclass (hunk 1 above)
Comment (DarthGandalf) [reply eclass/llvm.eclass]: Updated, PTAL

**Comment by grobian** [comment]
this looks ok to me, thanks!

File: sys-devel/llvm/files/llvm-16-darwin-curses.patch [hunk 2]
```diff
@@ -0,0 +1,15 @@
+On Darwin Prefix, don't try to link to /usr/lib/libcurses.dylib, but use Gentoo's libncurses.dylib
```
[the remark this answers is not in the document]
Comment (DarthGandalf) [reply sys-devel/llvm/files/llvm-16-darwin-curses.patch]: Updated.

File: eclass/llvm.eclass [hunk 3]
```diff
@@ -242,6 +249,10 @@ llvm_fix_tool_path() {
 llvm_pkg_setup() {
 	debug-print-function ${FUNCNAME} "${@}"
 
+	if [[ "${LLVM_ECLASS_SKIP_PKG_SETUP}" == yes ]]; then
```
Comment (mgorny) [inline eclass/llvm.eclass:252]:
```suggestion
	if [[ ${LLVM_ECLASS_SKIP_PKG_SETUP} ]]; then
```

File: eclass/llvm.eclass [hunk 4]
```diff
@@ -80,6 +80,13 @@ DEPEND="!!sys-devel/llvm:0"
 # Correct values of LLVM slots, newest first.
 declare -g -r _LLVM_KNOWN_SLOTS=( {18..8} )
 
+# @ECLASS_VARIABLE: LLVM_ECLASS_SKIP_PKG_SETUP
+# @INTERNAL
+# @DESCRIPTION:
+# If set to 'yes', llvm_pkg_setup will be noop.
```
Comment (mgorny) [inline eclass/llvm.eclass:86]:
```suggestion
# If set to a non-empty value, llvm_pkg_setup will not perform LLVM version check, nor set PATH.
```

**Pull request closed without merging**